### PR TITLE
feat(javascript): Reorganise and improve JS setup

### PR DESCRIPTION
**Pull request by talend-glorieux**
* Adds .npmignore
* Adds default .travis.yml
* Regroups all javascript related configuration files into a single
tools/javascript folder

### CODE REVIEW

**Comment by talend-glorieux** [comment]
@jsomsanith Could you take a look at this?

File: tools-javascript/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  - npm prune
+  - npm update
+after_success:
+  - 'curl -Lo travis_after_all.py https://git.io/travis_after_all'
+  - python travis_after_all.py
+  - export $(cat .to_export_back) &> /dev/null
+  - npm run semantic-release
+branches:
+  except:
+    - /^v\d+\.\d+\.\d+$/
```
Comment (jsomsanith) [inline tools-javascript/.travis.yml]: Question: what are those branches ? Releases ?

File: tools-javascript/.npmignore [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# Coverage directory used by tools like istanbul
+coverage
+
+# nyc test coverage
+.nyc_output
+
+# Grunt intermediate storage
+# (http://gruntjs.com/creating-plugins#storing-task-files)
+.grunt
```
Comment (jsomsanith) [inline tools-javascript/.npmignore]: Grunt ?

File: tools-javascript/.npmignore [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+pids
+*.pid
+*.seed
+
+# Directory for instrumented libs generated by jscoverage/JSCover
+lib-cov
+
+# Coverage directory used by tools like istanbul
+coverage
+
```
Comment (jsomsanith) [inline tools-javascript/.npmignore]: Souldn't we add build forlders ? (build/ dist/)

**Comment by jsomsanith** [comment]
Good idea to group the files into a javascript folder !

File: tools-javascript/.npmignore (hunk 2 above)
Comment (talend-glorieux) [inline tools-javascript/.npmignore]: I'll remove it ;)

File: tools-javascript/.npmignore (hunk 3 above)
Comment (talend-glorieux) [inline tools-javascript/.npmignore]: Yes! Good idea. Will do.

File: tools-javascript/.travis.yml (hunk 1 above)
Comment (talend-glorieux) [inline tools-javascript/.travis.yml]: Good question. I actually stole that configuration from react-cmf. @jmfrancois Any input?

Comment (acateland) [inline tools-javascript/.travis.yml]: to not run on tag

Comment (jmfrancois) [inline tools-javascript/.travis.yml]: @jsomsanith without this you will run into trouble with semantic release. AFAIK branch and tag are the same for travis. So if you push a tags, without this snippet, travis will run on it.

It's a good escape I think

**Pull request merged**